src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -422,18 +445,27 @@ abstract class AbstractCheckTask(
                 }
             }
         } else {
-            LOGGER.debug("Event tree id '{}' parent id '{}' is already published", rootEvent.id, parentEventID)
+            LOGGER.debug("Event tree id '{}' parent id '{}' is already published", refs.rootEvent.id, parentEventID)
         }
     }
 
     private fun completeEventOrReportError(prevState: State): Boolean {
         return try {
-            completeEvent(prevState)
-            doAfterCompleteEvent()
-            false
+            if (started) {
+                completeEvent(prevState)
+                doAfterCompleteEvent()
+                false
+            } else {
+                LOGGER.error("Check task was not started.")
+                refs.rootEvent.addSubEventWithSamePeriod()
+                    .name("Check failed: check task was not started.")
```

Comment:
Let's add extra information to help to understand the problem
```suggestion
                    .name("Check failed: task timeout elapsed before the check task was started. Please, check component resources for throttling or intensive GC")
```